### PR TITLE
Fix double free on OE_INVALID_IMAGE

### PR DESCRIPTION
**Pull request by thomasten**
If one tries to load an invalid ELF image, _unload_elf_image() is
called by _load_elf_image() and then again by
oe_load_elf_enclave_image(). Clear the struct in _unload_elf_image() so
that it does not call free on already freed pointers in the second
invocation.

### CODE REVIEW

**Review by anakrish — APPROVED** [review]
LGTM

@mrragava Can you also review this?

**Review by mingweishih — APPROVED** [review]
LGTM

**Comment by mrragava** [comment]
LGTM

File: tests/CMakeLists.txt [hunk 1]
```diff
@@ -29,6 +29,7 @@ if (OE_SGX)
   set(DEFINE_OE_SGX "-DOE_SGX")
 endif ()
 
+add_subdirectory(invalid_image)
```
Comment (mingweishih) [inline tests/CMakeLists.txt:32]: It turns out the the test is unix-specific. You would need to move the test as part of https://github.com/openenclave/openenclave/blob/dbd3e7ad0c0e4b374b3c63e860ca2698d9984d52/tests/CMakeLists.txt#L151

Comment (thomasten) [reply tests/CMakeLists.txt]: AArch64 fails, so I guess it's SGX-specific? I moved the test to OE_SGX and clarified the comment in the test. Please try again.

**Pull request merged**